operators/multiclusterobservability/controllers/placementrule/manifestwork.go

```diff
@@ -423,10 +457,135 @@ func createManifestWorks(
 
 	work.Spec.Workload.Manifests = manifests
 
-	err = createManifestwork(c, work)
+	if clusterName != clusterNamespace && os.Getenv("UNIT_TEST") != "true" {
+		// ACM 8509: Special case for hub/local cluster metrics collection
+		// install the endpoint operator into open-cluster-management-observability namespace for the hub cluster
+		log.Info("Creating resource for hub metrics collection", "cluster", clusterName)
+		err = createUpdateResourcesForHubMetricsCollection(c, manifests)
+	} else {
+		err = createManifestwork(c, work)
+	}
+
 	return err
 }
 
+func createCSR() ([]byte, []byte) {
+	keys, _ := rsa.GenerateKey(rand.Reader, 2048)
+
+	oidOrganization := []int{2, 5, 4, 11} // Object Identifier (OID) for Organization Unit
+	oidUser := []int{2, 5, 4, 3}          // Object Identifier (OID) for User
+
+	var csrTemplate = x509.CertificateRequest{
+		Subject: pkix.Name{
+			Organization: []string{"Red Hat, Inc."},
+			Country:      []string{"US"},
+			CommonName:   operatorconfig.ClientCACertificateCN,
+			ExtraNames: []pkix.AttributeTypeAndValue{
+				{Type: oidOrganization, Value: "acm"},
+				{Type: oidUser, Value: "managed-cluster-observability"},
+			},
+		},
+		DNSNames:           []string{"observability-controller.addon.open-cluster-management.io"},
+		SignatureAlgorithm: x509.SHA512WithRSA,
+	}
+	csrCertificate, _ := x509.CreateCertificateRequest(rand.Reader, &csrTemplate, keys)
+	csr := pem.EncodeToMemory(&pem.Block{
+		Type: "CERTIFICATE REQUEST", Bytes: csrCertificate,
+	})
+
+	privateKey := pem.EncodeToMemory(&pem.Block{
+		Type: "RSA PRIVATE KEY", Bytes: x509.MarshalPKCS1PrivateKey(keys),
+	})
+
+	return csr, privateKey
+}
+
+func createMtlsCertSecretForHubCollector(c client.Client) error {
```

Comment:
The overall procedure to create client certificate looks good. The client cert may expire, cert rotation is required. You can find some example code from registration agent: https://github.com/open-cluster-management-io/ocm/blob/1c3cb033b096c0e95575dfbf6d5d12126f5d7e57/pkg/registration/clientcert/cert_controller.go#L290